cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/main/java/com/devonfw/cobigen/openapiplugin/inputreader/OpenAPIInputReader.java

```diff
@@ -413,33 +415,53 @@ private PropertyDef extractReferenceProperty(OpenApi3 openApi, Schema componentS
      * @return list of {@link PathDef}'s
      */
     private List<PathDef> extractPaths(Map<String, ? extends Path> paths, String componentName) {
+        Matcher matcher;
+        Pattern pattern;
+        String match, root_component = null, version = null;
+
         List<PathDef> pathDefs = new LinkedList<>();
+
         for (String pathKey : paths.keySet()) {
             if (pathKey.toLowerCase().contains(componentName.toLowerCase())) {
                 String[] mp = pathKey.split("/");
                 String pathUri = "/";
-                for (int i = 3; i < mp.length; i++) {
-                    pathUri = pathUri.concat(mp[i] + "/");
+
+                try {
+                    match = "^\\/[^\\/]+\\/+[^\\/]+\\/(.+)";
+                    pattern = Pattern.compile(match);
+                    matcher = pattern.matcher(pathKey);
+                    matcher.find();
+                    pathUri += matcher.group(1);
+                    if (pathUri.charAt(pathUri.length() - 1) != "/".charAt(0)) {
```

Comment:
`"/".charAt(0)`

Wouldn't be easier to just compare

`"/"`

Therefore the `if` would look like this:

`if (pathUri.substring(pathUri.length() -1) != "/")`

Reply:
Unlike chars, `==` checks the object references. That's why I went for chars. I ll change this to String comparison but with the `equals()` function instead.

Reply:
Yeah, you are right, it should be `equals()`, I always forget it 👍